### ISSUE TITLE
Issue putting behind reverse proxy

### ISSUE THREAD

**Issue by adampaulberg** [RyanNoelk/OpenEats#312]
I was able to get the interface online, but when i attempt to login i get a "400 bad request" when it attempts to call "/api/v1/accounts/obtain-auth-token/". 

I changed in my docker-prod.yml the nginx port to "8000:80" since i have other things running on this machine. I want to reverse proxy port 8000 onto a subdomain 

//env-prod.yml
```
MYSQL_DATABASE=openeats
MYSQL_ROOT_PASSWORD=root

API_URL=0.0.0.0:8000
API_PORT=8000
DJANGO_SECRET_KEY=sdfsadfas32e98zsdvhhsnz6udvbksjdhfi4galshjfg
DJANGO_SETTINGS_MODULE=base.settings
DJANGO_DEBUG=False
ALLOWED_HOST='openeats.[mydomain]'
HTTP_X_FORWARDED_PROTO=true

NODE_ENV=production
NODE_URL=localhost:8080
NODE_PORT=8080
NODE_API_URL=http://openeats.[mydomain]
NODE_LOCALE=en
```
//nginx.conf
```
server {
    listen 80;
    server_name openeats.[mydomain];

    location / {
        proxy_pass http://debian:8000;
        proxy_set_header        Host            $host;
        proxy_set_header   X-Real-IP        $remote_addr;
        proxy_set_header   X-Forwarded-For  $proxy_add_x_forwarded_for;

    }
}
```

**Comment by RyanNoelk** [comment]
Two things:
1. Check to see if you can access the API directly via `openeats.[mydomain.com]/admin`
2. Remove the quotes around the allowed host env var. `ALLOWED_HOST='openeats.[mydomain]'` -> `ALLOWED_HOST=openeats.[mydomain]`

Let me know how those work :)

**Comment by adampaulberg** [comment]
I wasn't able to get to /admin, but attempting number 2 i did something that i'm always getting a 502 error now. I'll figure that out. 

I put quotes around the domain because your example shows quotes https://github.com/RyanNoelk/OpenEats/blob/master/docs/sample_env_file_for_docker.list

**Comment by adampaulberg** [comment]
I'm not sure what i am doing wrong. All i changed was ```ALLOWED_HOST=openeats.[mydomain]``` to not have quotes and reran ```docker-compose -f docker-prod.yml up -d```. I can no longer access the site and get a 502 error. After reverting the env_prod.yml back to having quotes, it's still broken.

**Comment by RyanNoelk** [comment]
I forgot to updated that example but there shouldn't be quotes in the ALLOWED_HOST var. (I'll update it tonight)

Try running `docker-compose -f docker-prod.yml up` without the -d flag. that will spit all of the logging to the console. If you are not able to get to the admin page then there might be something wrong in the way the API is setting itself up. Try and above command and see what the API is saying when it starts up.

**Comment by adampaulberg** [comment]
```
node_1   | npm info it worked if it ends with ok
node_1   | npm info using npm@5.3.0
node_1   | npm info using node@v8.6.0
node_1   | npm info lifecycle OpenEats@1.0.0~prestart: OpenEats@1.0.0
node_1   | npm info lifecycle OpenEats@1.0.0~start: OpenEats@1.0.0
node_1   |
node_1   | > OpenEats@1.0.0 start /code
node_1   | > if-env NODE_ENV=production && npm run start:prod || npm run start:dev
node_1   |
node_1   | npm info it worked if it ends with ok
node_1   | npm info using npm@5.3.0
node_1   | npm info using node@v8.6.0
node_1   | npm info lifecycle OpenEats@1.0.0~prestart:dev: OpenEats@1.0.0
node_1   | npm info lifecycle OpenEats@1.0.0~start:dev: OpenEats@1.0.0
node_1   |
node_1   | > OpenEats@1.0.0 start:dev /code
node_1   | > webpack-dev-server --hot --inline --content-base public --history-api-fallback
node_1   |
api_1    | Operations to perform:
api_1    |   Apply all migrations: admin, auth, authtoken, contenttypes, ingredient, list, news, recipe, recipe_groups, sessions, sites
api_1    | Running migrations:
api_1    |   No migrations to apply.
api_1    |   Your models have changes that are not yet reflected in a migration, and so won't be applied.
api_1    |   Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
node_1   | Project is running at http://0.0.0.0:8080/
node_1   | webpack output is served from /
node_1   | Content not from webpack is served from /code/public
node_1   | 404s will fallback to /index.html
api_1    |
api_1    | 0 static files copied to '/code/static-files', 84 unmodified.
api_1    | Starting OpenEats as root
api_1    | [2018-01-20 06:55:08 +0000] [10] [INFO] Starting gunicorn 19.6.0
api_1    | [2018-01-20 06:55:08 +0000] [10] [INFO] Listening at: http://0.0.0.0:8000 (10)
api_1    | [2018-01-20 06:55:08 +0000] [10] [INFO] Using worker: sync
api_1    | [2018-01-20 06:55:08 +0000] [15] [INFO] Booting worker with pid: 15
api_1    | [2018-01-20 06:55:08 +0000] [16] [INFO] Booting worker with pid: 16
api_1    | [2018-01-20 06:55:08 +0000] [17] [INFO] Booting worker with pid: 17
api_1    | [2018-01-20 06:55:09 +0000] [18] [INFO] Booting worker with pid: 18
api_1    | [2018-01-20 06:55:09 +0000] [19] [INFO] Booting worker with pid: 19
node_1   | Killed
node_1   | npm info lifecycle OpenEats@1.0.0~start:dev: Failed to exec start:dev script
node_1   | npm ERR! code ELIFECYCLE
node_1   | npm ERR! errno 137
node_1   | npm ERR! OpenEats@1.0.0 start:dev: `webpack-dev-server --hot --inline --content-base public --history-api-fallback`
node_1   | npm ERR! Exit status 137
node_1   | npm ERR!
node_1   | npm ERR! Failed at the OpenEats@1.0.0 start:dev script.
node_1   | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
node_1   |
node_1   | npm ERR! A complete log of this run can be found in:
node_1   | npm ERR!     /root/.npm/_logs/2018-01-20T06_55_32_681Z-debug.log
node_1   | npm info lifecycle OpenEats@1.0.0~start: Failed to exec start script
node_1   | npm ERR! code ELIFECYCLE
node_1   | npm ERR! errno 137
node_1   | npm ERR! OpenEats@1.0.0 start: `if-env NODE_ENV=production && npm run start:prod || npm run start:dev`
node_1   | npm ERR! Exit status 137
node_1   | npm ERR!
node_1   | npm ERR! Failed at the OpenEats@1.0.0 start script.
node_1   | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
node_1   |
node_1   | npm ERR! A complete log of this run can be found in:
node_1   | npm ERR!     /root/.npm/_logs/2018-01-20T06_55_32_706Z-debug.log
openeats_node_1 exited with code 137
```

**Comment by adampaulberg** [comment]
I got past this issue by updating node to version 8.9.4 as listed on https://github.com/mhart/alpine-node. Now unfortunately i am dealing with django not being able to connect to maria db 
``` 
django.db.utils.OperationalError: (1130, "Host '172.18.0.3' is not allowed to connect to this MariaDB server")

```

**Comment by RyanNoelk** [comment]
Hmm that is strange. The API shouldn't have any issues connecting to the DB. Are you using the docker images I've posted on dockerhub or are u building them yourself?

The database config is listed here: https://github.com/RyanNoelk/OpenEats/blob/master/api/base/settings.py#L27

Also it doesn't look like the nginx server is starting for you. Nginx should be siting in front of the django and node server to handle the redirects properly

**Comment by adampaulberg** [comment]
I'm not sure what I did to get it to work, but through deletion of the volumes and attempts to rebuild, eventually I was able to launch the site.

A couple things I've noticed. 
Creating a new recipe /news, I can either edit text or upload an image, I can't do both at the same time so I must edit these record to have an image 

Nginx configuration doesn't increase the max file size upload, so images over 1mb don't upload.
